This patch is based on the upstream commit described below, adapted for use
in the Debian package by Peter Michael Green.

commit 69e2db6efadf2d15514a8947f48acaf174024a57
Author: Dirkjan Ochtman <dirkjan@ochtman.nl>
Date:   Wed Oct 4 09:33:44 2023 +0200

    Upgrade to comrak 0.19

diff --git askama/src/filters/mod.rs askama/src/filters/mod.rs
index a7aa62b3b..f77e2f833 100644
--- askama/src/filters/mod.rs
+++ askama/src/filters/mod.rs
@@ -345,51 +345,22 @@ pub fn wordcount<T: fmt::Display>(s: T) -> Result<usize> {
 pub fn markdown<E, S>(
     e: E,
     s: S,
-    options: Option<&comrak::ComrakOptions>,
+    options: Option<&comrak::Options>,
 ) -> Result<MarkupDisplay<E, String>>
 where
     E: Escaper,
     S: AsRef<str>,
 {
-    use comrak::{
-        markdown_to_html, ComrakExtensionOptions, ComrakOptions, ComrakParseOptions,
-        ComrakRenderOptions, ListStyleType,
-    };
-
-    const DEFAULT_OPTIONS: ComrakOptions = ComrakOptions {
-        extension: ComrakExtensionOptions {
-            strikethrough: true,
-            tagfilter: true,
-            table: true,
-            autolink: true,
-            // default:
-            tasklist: false,
-            superscript: false,
-            header_ids: None,
-            footnotes: false,
-            description_lists: false,
-            front_matter_delimiter: None,
-        },
-        parse: ComrakParseOptions {
-            // default:
-            smart: false,
-            default_info_string: None,
-            relaxed_tasklist_matching: false,
-        },
-        render: ComrakRenderOptions {
-            escape: true,
-            // default:
-            hardbreaks: false,
-            github_pre_lang: false,
-            full_info_string: false,
-            width: 0,
-            unsafe_: false,
-            list_style: ListStyleType::Dash,
-            sourcepos: false,
-        },
-    };
-
-    let s = markdown_to_html(s.as_ref(), options.unwrap_or(&DEFAULT_OPTIONS));
+    use comrak::{markdown_to_html, Options};
+
+    let mut defaults = Options::default();
+    defaults.extension.strikethrough = true;
+    defaults.extension.tagfilter = true;
+    defaults.extension.table = true;
+    defaults.extension.autolink = true;
+    defaults.render.escape = true;
+
+    let s = markdown_to_html(s.as_ref(), options.unwrap_or(&defaults));
     Ok(MarkupDisplay::new_safe(s, e))
 }
 
Index: askama/Cargo.toml
===================================================================
--- askama.orig/Cargo.toml
+++ askama/Cargo.toml
@@ -43,7 +43,7 @@ version = "0.12.0"
 version = "0.10.3"
 
 [dependencies.comrak]
-version = "0.18"
+version = ">= 0.19, < 1.0"
 optional = true
 default-features = false
 
